src/dtx/dtx_common.c

```diff
@@ -267,48 +274,46 @@ dtx_leader_begin(struct dtx_id *dti, daos_unit_oid_t *oid, daos_handle_t coh,
 	int			dti_cos_count = 0;
 	int			i;
 
-	/* XXX: For leader case, we need to find out the potential
-	 *	conflict DTXs in the CoS cache, and append them to
-	 *	the dispatched RPC to non-leaders. Then non-leader
-	 *	replicas can commit them before real modifications
-	 *	to avoid availability trouble.
-	 */
-	if (!daos_is_zero_dti(dti)) {
-		dti_cos_count = vos_dtx_list_cos(coh, oid, dkey_hash,
-				intent == DAOS_INTENT_UPDATE ?
-				DCLT_PUNCH : DCLT_PUNCH | DCLT_UPDATE,
-				DTX_THRESHOLD_COUNT, &dti_cos);
-		if (dti_cos_count < 0)
-			return dti_cos_count;
-
-		if (dti_cos_count > 0 && dti_cos == NULL) {
-			/* There are too many conflict DTXs to be committed,
-			 * as to cannot be taken via the normal IO RPC. The
-			 * background dedicated DTXs batched commit ULT has
-			 * not committed them in time. Let's retry later.
-			 */
-			D_DEBUG(DB_TRACE, "Too many pontential conflict DTXs"
-				" for the given "DF_DTI", let's retry later.\n",
-				DP_DTI(dti));
-			return -DER_INPROGRESS;
-		}
-	}
-
 	dlh->dlh_handled_time = crt_hlc_get();
 	dlh->dlh_future = ABT_FUTURE_NULL;
 	D_ALLOC(dlh->dlh_subs, tgts_cnt * sizeof(*dlh->dlh_subs));
-	if (dlh->dlh_subs == NULL) {
-		if (dti_cos != NULL)
-			D_FREE(dti_cos);
+	if (dlh->dlh_subs == NULL)
 		return -DER_NOMEM;
-	}
 	for (i = 0; i < tgts_cnt; i++)
 		dlh->dlh_subs[i].dss_tgt = tgts[i];
 	dlh->dlh_sub_cnt = tgts_cnt;
 
 	if (daos_is_zero_dti(dti))
```

Comment:
We need to initialize dtx_handle::dth_xid for the case "if (daos_is_zero_dti(dti))". Otherwise, the subsequent checking of daos_is_zero_dti(dtx_handle::dth_xid) may get confused result, such as in the dtx_leader_end().

Reply:
OK, I'll zero it.
In the code the passed in dlh is inited as "struct dtx_leader_handle	dlh = { 0 };", so should no the un-init issue. But I agree that better to explicit zeor it for zero dti here. thanks.